File: exercise/week04/c#/Construct/StealthCloakUpdater.cs
namespace Construct;

public class StealthCloakUpdater : IArtefactUpdater
{
    public void Update(Artefact artefact)
    {
        artefact.TimeToLive--;
        if (artefact.TimeToLive % 2 == 0) //lose integrity every 2 days
            artefact.Integrity = Math.Max(0, artefact.Integrity - 1);
    }
}

File: exercise/week04/php/src/Construct/StealthCloakUpdater.php
<?php

namespace Construct;

class StealthCloakUpdater implements ArtefactUpdater
{
    public function update(Artefact $artefact): void
    {
        $artefact->timeToLive--;
        if ($artefact->timeToLive % 2 === 0) {
            $artefact->integrity = max(0, $artefact->integrity - 1);
        }
    }
}